pkg/apis/provisioning/v1alpha1/provisioner_validation.go

```diff
@@ -12,25 +12,122 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// +kubebuilder:webhook:path=/validate-provisioning-karpenter-sh-v1alpha1-provisioner,mutating=false,sideEffects=None,failurePolicy=fail,groups=provisioning.karpenter.sh,resources=provisioners,verbs=create;update,versions=v1alpha1,name=vprovisioner.kb.io
+
 package v1alpha1
 
 import (
+	"fmt"
+
+	f "github.com/awslabs/karpenter/pkg/utils/functional"
```

Comment:
Any necessity to use `f` as opposed `functional` here?

Reply:
Rebasing.